### PR TITLE
add travis and reconfigure NB for fun and profit. 

### PR DESCRIPTION
**Pull request by drunkirishcoder**

### CODE REVIEW

**Comment by seancribbs** [comment]
We need to remove the `etcdv3-rs` dependency so this will build.

**Comment by zeeshanlakhani** [comment]
@drunkirishcoder @seancribbs were you guys going to remove the vagrant stuff, and update utils/sandbox to incorporate NB all at once?

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# -*- mode: ruby -*-
```
Comment (zeeshanlakhani) [inline Vagrantfile]: so, just need to remove this now, yeah?

Comment (drunkirishcoder) [reply Vagrantfile]: yep. removed this and `vm-setup.sh`.

Comment (drunkirishcoder) [reply Vagrantfile]: Can't delete `vm-setup.sh`. Travis needs it.

File: docker.mk [hunk 2]
```diff
@@ -8,31 +8,57 @@ TAG ?= latest
...
+         -v /sys/kernel/mm/hugepages:/sys/kernel/mm/hugepages \
+         -v /sys/devices/system/node:/sys/devices/system/node \
+         -v /sbin/modinfo:/sbin/modinfo \
+         -v /bin/kmod:/bin/kmod \
+         -v /sbin/lsmod:/sbin/lsmod \
+         -v /mnt/huge:/mnt/huge \
+         -v /dev:/dev \
+         -v /var/run:/var/run
+
+ALL_MOUNTS = $(MOUNTS) \
```
Comment (zeeshanlakhani) [inline docker.mk]: weird spacing stuff btw.

File: scripts/travis-setup.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+echo "hugetlbfs /mnt/huge hugetlbfs rw,mode=0777 0 0" >> /etc/fstab
+
+# Install the uio_pci_generic driver
+modprobe uio_pci_generic
+
+# Load modules at boot
+echo "uio" >> /etc/modules
+echo "uio_pci_generic" >> /etc/modules
+
+echo "cd /vagrant" >> /home/vagrant/.bashrc
```
Comment (zeeshanlakhani) [inline scripts/travis-setup.sh]: can remove this.

**Comment by seancribbs** [comment]
:shipit:

**Pull request merged**